### ISSUE TITLE
Jaggies on game icons

### ISSUE THREAD

**Issue by Org-gith** [renpy/renpy#4957]
This isn't important, just a small something I noticed:
![image](https://github.com/renpy/renpy/assets/78823063/e00ac92a-0051-4a75-8ba1-0dc6d6980414)
The icon for Ren'py is 250x250 which is big and results in a pixellated icon when downsized by the OS. Maybe something closer to the target would diminish the jaggies.

**Comment by renpytom** [comment]
Having tried it out, it doesn't seem to help. I believe we'd need new SDL APIs to address this.

**Comment by mal** [comment]
Good icon construction is an art in itself, and while Ren'Py provides a basic version of this for convenience, a better option for those wanting to tailor their icons to a greater degree is to create and manage the `ico` (win) and `icns` (mac) files themselves, per <https://www.renpy.org/doc/html/build.html#special-files>. That way you have full control not only of the primary image quality, but also of the various resolution steps.

**Comment by bober-gith** [comment]
Definitely, it's something that requires attention at the creator level, but the link for building icons is hmm the icon of the actual window of the game, but windows os displays not that one in the taskbar but "window_icon.png" inside game/gui (I guess something similar is happening for the Ren'py launcher itself). 
It's solvable right now by making "window_icon.png" smaller in size, however uncertain how it behaves on another os to have it much lower without an alternative (if it's called window_icon, maybe that means only used by windows, so maybe that's no problem?). I think when you have a sort of embedded "icon" with precise dimensions this usually leads the creator to respect that dimension, which can lead to most Ren'py games having that problem inadvertently on windows os.
This is very much not an important issue, just to explain why I opened this thing in the first place.

**Comment by hsandt** [comment]
Weird, I see the post is from 2023 yet I've been working on a few Renpy games back then and I don't remember having this issue on Linux.

Today I just ran Renpy 8.4.1 (the app, not even a distributed game) on Windows and the Renpy application icon itself is jagged/pixelated.

<img width="103" height="51" alt="Image" src="https://github.com/user-attachments/assets/56eeb2a9-2592-44fd-9e84-0da54e34106c" />

The game app besides is uniform purple and smaller so I can't really say if it's good or not.

So it may be a more generic issue about how Renpy exports icons for Windows? I understand that devs should carefully craft their own game icons, but the Renpy icon itself should be correct.

EDIT: I just got a runtime error and the error popup icon looked like this rightmost icon:

<img width="139" height="51" alt="Image" src="https://github.com/user-attachments/assets/2ad5e442-ff02-47e1-961c-e1cce21f3a33" />

which seems to use some bilinear or other smooth interpolation instead of nearest neighbor! And just besides you can still see the SDK app icon! Why the difference?

**Comment by hsandt** [comment]
Today, on Windows 11, I installed Renpy 8.5.3 and launched it for the first time, the icon was jaggy. Then I right-clicked > Pin to taskbar and it was immediately replaced with the cleaner "bilinear-scaled" icon. Maybe it forced the taskbar to retrieve the proper icon and/or apply the proper interpolation?
